RevenueCatUI/CustomerCenter/Data/Transaction.swift

```diff
@@ -24,6 +24,12 @@ protocol Transaction {
     var price: ProductPaidPrice? { get }
     var periodType: PeriodType { get }
     var purchaseDate: Date { get }
+    var unsubscribeDetectedAt: Date? { get }
+    var billingIssuesDetectedAt: Date? { get }
+    var gracePeriodExpiresDate: Date? { get }
+    var refundedAtDate: Date? { get }
+    var storeIdentifier: String? { get }
+    var identifier: String? { get }
```

Comment:
I have been wondering if we should move some of these improvements to the `CustomerInfo`... but I guess that can be a separate PR if needed.

Reply:
I've been thinking on renaming this to something like `AnyPurchase` to disambiguate it from the existing `Transaction` struct. I can open a draft after I wrap this up and see how it looks

Reply:
No hurry, I just thought trying to join the different transaction types into one like this could be useful... But not a problem for this PR.

Also, personally I would prefer something more related to `Transaction` than to Purchase... Mostly because `Purchase` as a term already has a lot of meanings.